internal/controller/purge_controller.go

```diff
@@ -45,55 +46,51 @@ type RemoteClientResolver func(context.Context, client.ObjectKey) (client.Client
 type PurgeReconciler struct {
 	client.Client
 	record.EventRecorder
+	logger                logr.Logger
```

Comment:
Why is this change needed? This field isn't even initialized upon construction, but assigned every reconciliation now. You can either pass the logger to functions as a parameter where you moved the log statements to, or `logf.FromContext(ctx)` within them, or inject it when creating PurgeReconciler in main setup.

Reply:
Reverted